packages/blade/src/components/Input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,227 @@
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'done';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType = autoCompleteSuggestionType ?? 'none';
+  } else if (type === 'search') {
+    keyboardAndAutocompleteProps.keyboardType = 'search';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'search';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType = autoCompleteSuggestionType ?? 'none';
+  }
+
+  return keyboardAndAutocompleteProps;
+};
+
+// need to do this to tell TS to infer type as TextInput of React Native and make it believe that `ref.current.clear()` exists
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isReactNative = (_textInputRef: any): _textInputRef is TextInputReactNative => {
+  return getPlatformType() === 'react-native';
+};
+
+export const TextInput = ({
+  label,
+  labelPosition = 'top',
+  placeholder,
+  type = 'text',
+  defaultValue,
+  name,
+  value,
+  maxCharacters,
+  onChange,
+  onBlur,
+  isDisabled,
+  necessityIndicator,
+  validationState,
+  errorText,
+  helpText,
+  successText,
+  isRequired,
+  icon,
+  prefix,
+  showClearButton,
+  onClearButtonClick,
+  isLoading,
+  suffix,
+  autoFocus,
+  keyboardReturnKeyType,
+  autoCompleteSuggestionType,
+}: TextInputProps): ReactElement => {
+  const textInputRef = React.useRef<HTMLInputElement | TextInputReactNative>(null);
+  const [shouldShowClearButton, setShouldShowClearButton] = useState(false);
+
+  const renderInteractionElement = (): ReactNode => {
+    if (isLoading) {
+      return <InfoIcon size="medium" color="surface.text.subtle.lowContrast" />;
+    }
+
+    if (shouldShowClearButton) {
+      return (
+        <IconButton
+          size="medium"
+          icon={CloseIcon}
+          onClick={() => {
+            if (isEmpty(value) && textInputRef.current) {
+              // when the input field is uncontrolled take the ref and clear the input and then call the onClearButtonClick function
+              if (isReactNative(textInputRef.current)) {
+                textInputRef.current.clear();
+              } else if (textInputRef.current instanceof HTMLInputElement) {
+                textInputRef.current.value = '';
+              }
+            }
+            // if the input field is controlled just call the click handler and the value change shall be left upto the consumer
+            onClearButtonClick?.();
+            setShouldShowClearButton(false);
+          }}
+          accessibilityLabel="Clear Input Content"
+        />
+      );
+    }
+
+    return null;
+  };
+
+  return (
+    <BaseInput
+      id="textinput"
+      ref={textInputRef as React.Ref<HTMLInputElement>}
+      label={label}
+      labelPosition={labelPosition}
+      placeholder={placeholder}
+      defaultValue={defaultValue}
+      value={value}
+      name={name}
+      maxCharacters={maxCharacters}
```

Comment:
accessibility good to have feature: 

Announce how many characters are left, Right now SR users have no idea. 

Let's track this on an issue, we can do it as an enhancement.